### PR TITLE
options/linux: Add mallopt

### PR DESCRIPTION
**Pull request by Dennisbonke**
Part of the mlibc LFS project.

### CODE REVIEW

**Comment by 64** [comment]
What needs this?

File: options/linux/generic/malloc.cpp [hunk 1]
```diff
@@ -10,3 +10,7 @@ void *memalign(size_t, size_t) {
 	__ensure(!"Not implemented");
 	__builtin_unreachable();
 }
+
+int mallopt(int, int) {
+	return 1;
```
Comment (ArsenArsen) [inline options/linux/generic/malloc.cpp:15]: Return zero instead. This didn't succeed (it isn't implemented). Funnily enough, errno isn't set by this function, which doesn't even let us mention that it's not implemented via ENOSYS. (Also, wtf? I've never seen this function in actual use)

**Comment by 64** [comment]
I agree it doesn't make sense for us to implement this. It should be easy to patch programs which need it.

**Comment by ArsenArsen** [comment]
I don't think it's accurate that this is an implementation detail, though. It's definitely a public API, just one we don't really have a match for.

**Comment by qookei** [comment]
It seems like more than just glibc ever implemented it:
```
CONFORMING TO
       This  function is not specified by POSIX or the C standards.  A similar
       function exists on many System V derivatives, but the range  of  values
       for  param  varies  across systems.  The SVID defined options M_MXFAST,
       M_NLBLKS, M_GRAIN, and M_KEEP, but only the first of  these  is  imple‐
       mented in glibc.
```

**Comment by Geertiebear** [comment]
Maybe just add a warning that this is not implemented.

**Comment by Geertiebear** [comment]
I think we've reached a consensus on discord that we are dropping this change, unless we get a use for it in mlibc. Closing.

**Pull request closed without merging**